radix-engine/src/kernel/call_frame.rs

```diff
@@ -522,27 +546,31 @@ impl<L: Clone> CallFrame<L> {
             let mut new_owned_nodes: IndexSet<NodeId> = index_set_new();
             for own in substate.owned_nodes() {
                 if !new_owned_nodes.insert(own.clone()) {
-                    return Err(CloseSubstateError::ContainsDuplicatedOwns);
+                    return Err(CallbackError::Error(
+                        CloseSubstateError::ContainsDuplicatedOwns,
+                    ));
                 }
             }
             for own in &new_owned_nodes {
                 if !substate_lock.owned_nodes.contains(own) {
                     // Node no longer owned by frame
                     self.take_node_internal(own)
-                        .map_err(CloseSubstateError::TakeNodeError)?;
+                        .map_err(|e| CallbackError::Error(CloseSubstateError::TakeNodeError(e)))?;
```

Comment:
I should probably implement `From` on these conversion, we can just `?`/`.into()`.